### DISCUSSION TITLE
Unicode characters on C4 display

### DISCUSSION THREAD

**Discussion by BilldarBagdar** [markusschloesser/MackieC4_P3#34] [Q&A] [answered]
Have you seen, for example, the "umlaut vowels" you added to those methods in EncoderController actually appear on the LCD screens of the C4?  If so, I'm kind of surprised Mackie went to the effort of supporting unicode characters like that, especially with all the cutstom xml and sysex they were implementing.  A lot of companies would have chosen to strictly stick with single byte character representations when designing that kind of communications system.

On the other hand, I guess I shouldn't be surprised at all.  I can easily imagine you have synths that show "unicode patch names" on their LCDs, so why wouldn't a company like Mackie have wanted to support such non-US markets with the C4?

**Comment by markusschloesser** [comment]
just checked, no unicode on either MCU nor C4. They are transformed to spaces (MCU) or "#" (C4). so we might just filter them anyway.
Tried with track names

**Comment by BilldarBagdar** [comment, marked as answer]
If only single byte characters are supported by the C4 for display on the LCDs, then we should filter out (or translate) ALL unicode characters to single byte character text going to the LCD displays.  I know that is a very "English speaking" perspective, but we didn't select the design, and if such characters won't display properly anyway, they should not be supported.

I mean it will be easier to say, for example, remove all characters with byte values less than 0x20 or greater than 0x7F (and/or multi-byte values) than to explicitly call out all the individual characters to remove.

**Reply by markusschloesser** [reply]
https://github.com/markusschloesser/MackieC4_P3/blob/92ad2c62f68ef8e76b9d4279a0643bf6f7fb04e4/wip/MackieC4/EncoderController.py#L1402  
;-)
